src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -777,15 +778,21 @@ public void terminateTask(String taskId, TerminateInfo terminateInfo) {
     }
 
     @Transactional(propagation = Propagation.REQUIRES_NEW)
-    public TaskResource updateTaskIndex(String taskId) {
-        Optional<TaskResource> findTaskResponse = cftTaskDatabaseService.findByIdAndObtainPessimisticWriteLock(taskId);
-        if (findTaskResponse.isPresent()) {
-            TaskResource taskResource = findTaskResponse.get();
-            taskResource.setIndexed(true);
+    public void updateTaskIndex(String taskId, boolean indexed) {
+        try {
+            Optional<TaskResource> findTaskResponse = cftTaskDatabaseService
+                .findByIdAndWaitAndObtainPessimisticWriteLock(taskId);
 
-            return cftTaskDatabaseService.saveTask(taskResource);
-        } else {
-            throw new TaskNotFoundException(TASK_NOT_FOUND_ERROR);
+            if (findTaskResponse.isPresent()) {
+                TaskResource taskResource = findTaskResponse.get();
+                taskResource.setIndexed(indexed);
+
+                cftTaskDatabaseService.saveTask(taskResource);
+            } else {
+                throw new TaskNotFoundException(TASK_NOT_FOUND_ERROR);
+            }
+        } catch (LockTimeoutException ex) {
```

Comment:
I think RWA-2332 will address this, but we can possibly get other Sql exceptions here.

Reply:
Changed to handle PersistenceException